### DISCUSSION TITLE
Updated Casks

### DISCUSSION THREAD

**Discussion by mattpackwood** [Homebrew/discussions#3337] [Casks] [answered]
Is there a way to see updated casks with the new 3.5.0?
I have used --greedy in the past to get apps that can update themselves updated remotely but with the new update functionality I no longer get the list of updates with the "star" showing a cask update is available.

**Comment by SMillerDev** [comment]
Hoping that @apainintheneck can answer this one

**Comment by apainintheneck** [comment]
Which brew command are you using to check for updates? I assume you're talking about `brew outdated`. Is that right?

Also, what types of casks are we talking about? The `--greedy` flag includes both `auto_update true` or `version :latest` casks. If you're not sure, could you name a few you'd expect to see in the list.

**Comment by mattpackwood** [comment]
Hi @apainintheneck ,

My workflow is to run `brew update`
then I see which casks have "*" next to them (they have been updated)
Then I run `brew upgrade name_of_cask`
Then I run `brew upgrade` to get any formulae

Matt

**Reply by apainintheneck** [reply]
Does `brew outdated` show the casks that you expect to see? That would help narrow down whether it's generally related to casks or the update command in particular.

I think what's happening is that the default behavior of `brew update` was changed to only list installed formulae and casks that were updated. Plus, it stays silent instead of printing an empty list of updated formulae or casks.

For example, I ran `brew update` recently so when I ran it again all I got was the following.
```bash
~ ❯❯❯ brew update
Updated 1 tap (homebrew/core).
```

**Reply by apainintheneck** [reply]
Also, you can restore the old behavior by setting the `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE` environment variable. There is currently a bug being worked on with it not printing new casks and formulae but modified ones show up just fine which should work for your use case.

More information can be found in this thread. #3353

**Comment by mattpackwood** [comment]
I have been doing some testing.

I have seen that the following ARE reported as being out of date / updates available:
```
awscli (2.7.6) < 2.7.7
gh (2.11.3) < 2.12.1
1password-cli (2.4.0) != 2.4.1
multipass (1.9.1) != 1.9.2
```
I was watching the RSS Feed for cask updates and saw that GitHub desktop was updated.  When I set up `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE=TRUE` and ran `brew update` I still only saw the 4 listed above.

I will experiment more in the next few days.

**Comment by mattpackwood** [comment]
@apainintheneck 

It takes me a few days for each check as I have to wait until casks are updated.

What I am seeing at the moment is:
```
Matt-Packwoods-iMac:~ $ brew update                    
Updated 2 taps (homebrew/core and homebrew/cask).
==> Outdated Casks
gimp

You have 1 outdated cask installed.
You can upgrade it with brew upgrade
or list it with brew outdated.
Matt-Packwoods-iMac:~ $ brew outdated                  
gimp (2.10.30) != 2.10.32
Matt-Packwoods-iMac:~ $ brew info homebrew/cask/termius
termius: 7.42.1 (auto_updates)
https://www.termius.com/
/opt/homebrew/Caskroom/termius/7.42.0 (121B)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/termius.rb
==> Name
Termius
==> Description
SSH client
==> Artifacts
Termius.app (App)
==> Analytics
install: 259 (30 days), 839 (90 days), 2,910 (365 days)
```

So Homebrew knows that Termius is out of date but is silent about it as it is an "auto_update" app.

**Reply by Bo98** [reply]
`brew outdated` does take a `--greedy` switch you can use, so if you're checking the output of that you'll need to use that flag if you want to see Termius.

There is indeed no way currently to have that in `brew update`'s report, with the closest alternative being `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE` which lists _all_ modifications to Casks regardless if they are installed.

**Comment by mattpackwood** [comment, marked as answer]
@apainintheneck No worries, I will work through it.
